sw/device/silicon_creator/manuf/base/sram_cp_provision.c

```diff
@@ -61,99 +62,102 @@ static void manually_init_ast(uint32_t *data) {
   }
 }
 
-static status_t flash_info_page_0_erase(void) {
-  uint32_t byte_address = 0;
-  // DeviceId and ManufState are located on the same flash info page.
-  TRY(flash_ctrl_testutils_info_region_setup_properties(
-      &flash_ctrl_state, kFlashInfoFieldDeviceId.page,
-      kFlashInfoFieldDeviceId.bank, kFlashInfoFieldDeviceId.partition,
-      kFlashInfoPage0Permissions, &byte_address));
-  TRY(flash_ctrl_testutils_erase_page(&flash_ctrl_state, byte_address,
-                                      kFlashInfoFieldDeviceId.partition,
-                                      kDifFlashCtrlPartitionTypeInfo));
-  return OK_STATUS();
-}
-
-static status_t flash_info_page_0_write(
-    manuf_cp_provisioning_data_t *provisioning_data) {
-  uint32_t byte_address = 0;
+static status_t flash_info_page_0_read_and_validate(
+    manuf_cp_provisioning_data_out_t *console_out) {
   TRY(flash_ctrl_testutils_info_region_setup_properties(
-      &flash_ctrl_state, kFlashInfoFieldDeviceId.page,
-      kFlashInfoFieldDeviceId.bank, kFlashInfoFieldDeviceId.partition,
-      kFlashInfoPage0Permissions, &byte_address));
-
-  // Write DeviceId.
-  TRY(flash_ctrl_testutils_write(
-      &flash_ctrl_state, byte_address, kFlashInfoFieldDeviceId.partition,
-      provisioning_data->device_id, kDifFlashCtrlPartitionTypeInfo,
-      kHwCfgDeviceIdSizeIn32BitWords));
-
-  // Write ManufState (on same page as DeviceId).
-  TRY(flash_ctrl_testutils_write(
-      &flash_ctrl_state, byte_address + kFlashInfoFieldManufState.byte_offset,
-      kFlashInfoFieldManufState.partition, provisioning_data->manuf_state,
-      kDifFlashCtrlPartitionTypeInfo, kHwCfgManufStateSizeIn32BitWords));
-
-  // Write AST calibration values (on same page as DeviceId).
-  TRY(flash_ctrl_testutils_write(
-      &flash_ctrl_state,
-      byte_address + kFlashInfoFieldAstCalibrationData.byte_offset,
-      kFlashInfoFieldAstCalibrationData.partition, ast_cfg_data,
-      kDifFlashCtrlPartitionTypeInfo,
+      &flash_ctrl_state, kFlashInfoFieldCpDeviceId.page,
+      kFlashInfoFieldCpDeviceId.bank, kFlashInfoFieldCpDeviceId.partition,
+      kFlashInfoPage0Permissions, /*offset=*/NULL));
+
+  // Read (wafer) lot name.
```

Comment:
@benjbender if you could enlighten me on the encoding of the lot name, I could address the "TODO" below on line 114, and validate the second word of the CP device ID contains the correct "year", "week", and "lot number" data. Thanks!

Reply:
Synced with @benjbender offline to confirm the format of this; I have updated the flow to write the CP device ID to flash so it will not be expected to be there already.